tests/rules/expect-type.test.ts

```diff
@@ -6,11 +6,11 @@ import { name, rule } from '../../src/rules/expect';
 
 // Valid test cases.
 const valid: ReadonlyArray<ValidTestCase> = [
-  // Primitive type
+  // Twoslash
   {
     code: dedent`
-      // $ExpectType number
-      const t = 6 as number;
+      const t = { a: 17, b: 'on' as const };
+      //    ^? { a: number; b: "on"; }
```

Comment:
In the Playground, twoslash would show:

```ts
      const t = { a: 17, b: 'on' as const };
      //    ^? const t: { a: number; b: "on"; }
```

(note the additional `const t: `)

I think this display is preferable, but looking at the [code](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/ts-twoslasher/src/index.ts#L522-L553) that @orta referenced in https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type/issues/43#issuecomment-1073545192 (thanks, Orta!) this uses the language service to get quick info. Is that something I can access in an eslint rule?

Reply:
[A quick look](https://github.com/microsoft/TypeScript/blob/main/src/services/services.ts#L1698-L1735) shows the main chunk of work happens in an `@internal` section of the codebase `SymbolDisplay`: 

```ts
            const { symbolKind, displayParts, documentation, tags } = typeChecker.runWithCancellationToken(cancellationToken, typeChecker =>
                SymbolDisplay.getSymbolDisplayPartsDocumentationAndSymbolKind(typeChecker, symbol, sourceFile, getContainerNode(nodeForQuickInfo), nodeForQuickInfo)
            );
```

Assuming you have a type checker instance, and the nodes - you'd have enough data around to call it - it just isn't (publicly) available in outside of calling it via the service call

Reply:
Thanks @orta. It looks like `getSymbolDisplayPartsDocumentationAndSymbolKind` is exactly the function I want, but [holy cow is it a lot of code][1] and it's not really clear to me how I can access it.

If either of you know an eslint plugin that uses the language service, or know if this is Not Allowed, that would be very helpful! Some googling turned up [this plugin][2], but it's hard for me to tell if this is a sanctioned approach to getting a language service.

[1]: https://github.com/microsoft/TypeScript/blob/0d5abd8a15fb8bbb97e8f53cdf4d28299516ff8c/src/services/symbolDisplay.ts#L138-L709
[2]: https://github.com/Quramy/typescript-eslint-language-service/blob/master/src/eslint-adapter.ts

Reply:
I'm not at a computer with access to the ts repo, but from memory creating a langauge service isn't too expensive and doesn't have to represent the long-running server aspects used in IDE experiences (e.g. twoslash doesn't do that )

I don't know anything about ts eslint though, but it could also make sense for it to have a singleton service reference

Reply:
You're right, making a language service turns out to be easier than I was expecting. It's only working on a single file at a time, but this is promising!

![expect-type-fix](https://user-images.githubusercontent.com/98301/162592605-184fe6e5-e069-4a63-aa87-387f4e1b85df.gif)

Reply:
Beuatiful